packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -141,13 +145,40 @@ describeNoCompat.skip("Batching failures", (getTestObjectProvider) => {
             });
         await runAndValidateBatch(provider, proxyDsf, this.timeout());
     });
-    it("working batch",
-    async function() {
+
+    it("working batch", async function() {
         const provider = getTestObjectProvider({ resetAfterEach: true });
         await runAndValidateBatch(provider, provider.documentServiceFactory, this.timeout());
     });
+
+    it("contains batch metadata", async function() {
```

Comment:
does this test still pass if we run it 1000 times? the others only failed occasionally

Reply:
i believe the hypothesis was that this would only fail is specific scenarios, like maybe reconnect, so that's why we see inconsistent failures

Reply:
Yes, this test passes even when run 1000 times (I also ran multiple times for good measure since the issues appeared ~3 times per 1000).
I couldn't pinpoint exactly where the issue was originally coming from in my old version of next, but I did see that the issue was no batch metadata was being applied in the failure cases. It has something to do with replaying ops, but I couldn't get it to catch while debugging due to some things with reconnecting in the ConnectionManager.